lib/web/mage/sticky.js

```diff
@@ -41,8 +44,34 @@ define([
 
             if (!isStatic && this.element.is(':visible')) {
                 offset = $(document).scrollTop() - this.parentOffset;
+
+                if (typeof this.options.spacingTop === 'function') {
+                    offset += this.options.spacingTop();
+                } else {
+                    offset += this.options.spacingTop;
+                }
+
                 offset = Math.max(0, Math.min(offset, this.maxOffset));
-                this.element.css('top', offset);
+
+                if (offset &&
+                    this.options.offsetTop &&
+                    !this.element.hasClass(this.options.stickyClass)) {
```

Comment:
Can you explain the desired logic here? It's hard to tell from the code what `!this.element.hasClass(this.options.stickyClass)` means for the business logic. Does it mean "the element is currently displayed as sticky" or "the element is NOT currently displayed as sticky"? For more readability, you could assign it to a variable `stuck`.